src/runtime/types.cpp

```diff
@@ -919,13 +919,18 @@ Box* instancemethodGet(BoxedInstanceMethod* self, Box* obj, Box* type) {
         return self;
     }
 
-    // TODO subclass test
+    RELEASE_ASSERT(PyType_Check(type), "need to handle this for old-style types too");
+    RELEASE_ASSERT(PyType_Check(self->im_class), "need to handle this for old-style types too");
+    if (!isSubclass(static_cast<BoxedClass*>(type), static_cast<BoxedClass*>(self->im_class))) {
+        return self;
+    }
 
-    return new BoxedInstanceMethod(obj, self->func);
+    return new BoxedInstanceMethod(obj, self->func, self->im_class);
 }
 
 Box* instancemethodNew(BoxedClass* cls, Box* func, Box* self, Box** args) {
     Box* classObj = args[0];
+    RELEASE_ASSERT(PyType_Check(classObj), "need to check this and throw an exception");
```

Comment:
is this right?  this (and the cast to BoxedClass a little bit later) seems unnecessary